[File][m] process non-utf8 encoding files

AcckiyGerman opened the pull request:
This changes allows  `FileLocal` class to decode non-utf8 files into stream, so the `data cat` now shows proper characters for people (https://github.com/datahq/datahub-qa/issues/105).

data.js issue: https://github.com/datahq/data.js/issues/38

Testing:
- `data cat iso8859.csv` - you should see the proper letters

AcckiyGerman committed:
[File][m] process non-utf8 encoding files #issues:
- https://github.com/datahq/data.js/issues/38
- https://github.com/datahq/datahub-qa/issues/105

AcckiyGerman committed:
Merge branch 'master' into fix/non-utf8-encoding

zelima commented on lib/index.js:
```diff
@@ -69,7 +72,13 @@ class File {
+      // and then 'stream-to-array' lib will return strings instead of Buffer objects.
+      // Let's transform such strings, otherwise `Buffer.concat` will cause an exception
+      return Buffer.concat(buffers.map(buffer => {
+        if (typeof buffer === 'string'){return Buffer(buffer)}
```
Let's avoid one-liner curly brackets

zelima commented on lib/index.js:
```diff
@@ -69,7 +72,13 @@ class File {
+      // Let's transform such strings, otherwise `Buffer.concat` will cause an exception
+      return Buffer.concat(buffers.map(buffer => {
+        if (typeof buffer === 'string'){return Buffer(buffer)}
+        else {return buffer}
```
Same here - no one-liner curly brackets

zelima reviewed (approved):
Looks fine to me. just minor comments. But I strongly recommend @anuveyatsu to review before merging

AcckiyGerman committed:
[File][m] process non-utf8 encoding files #issues:
- https://github.com/datahq/data.js/issues/38
- https://github.com/datahq/datahub-qa/issues/105

AcckiyGerman commented on lib/index.js:
```diff
@@ -109,7 +120,19 @@ class FileLocal extends File {
+     * But anyway this could happen only with non-utf8 files, which are broken without this fix.
+     */
+
+    // non utf-8 files are decoded by iconv-lite module
```
test

anuveyatsu reviewed (approved):
Looks good to me! Will think about FileRemote class..
